app/services/capital_collator_and_assessor.rb

```diff
@@ -1,31 +1,45 @@
 class CapitalCollatorAndAssessor
+  PersonCapitalCollation = Struct.new(:total_vehicle)
+  CapitalCollation = Struct.new(:applicant_capital_collation, :partner_capital_collation)
   class << self
     def call(assessment)
-      data = Collators::CapitalCollator.call(
-        submission_date: assessment.submission_date,
-        capital_summary: assessment.capital_summary,
-        maximum_subject_matter_of_dispute_disregard: maximum_subject_matter_of_dispute_disregard(assessment),
-        pensioner_capital_disregard: pensioner_capital_disregard(assessment),
-      )
-      assessment.capital_summary.update!(data)
+      data = collate_applicant_capital(assessment)
+      assessment.capital_summary.update!(data.except(:total_vehicle))
       if assessment.partner.present?
-        partner_data = Collators::CapitalCollator.call(
-          submission_date: assessment.submission_date,
-          capital_summary: assessment.partner_capital_summary,
-          pensioner_capital_disregard: 0,
-          maximum_subject_matter_of_dispute_disregard: 0,
-        )
-        assessment.partner_capital_summary.update!(partner_data)
+        partner_data = collate_partner_capital(assessment)
+        assessment.partner_capital_summary.update!(partner_data.except(:total_vehicle))
         assessment.capital_summary.update!(combined_assessed_capital: assessment.capital_summary.assessed_capital +
                                                                         assessment.partner_capital_summary.assessed_capital)
       else
         assessment.capital_summary.update!(combined_assessed_capital: assessment.capital_summary.assessed_capital)
       end
       Assessors::CapitalAssessor.call(assessment.capital_summary, assessment.capital_summary.combined_assessed_capital)
+      CapitalCollation.new(
+        PersonCapitalCollation.new(data[:total_vehicle]),
+        (PersonCapitalCollation.new(partner_data[:total_vehicle]) if assessment.partner.present?),
+      )
     end
 
   private
 
+    def collate_applicant_capital(assessment)
+      Collators::CapitalCollator.call(
+        submission_date: assessment.submission_date,
+        capital_summary: assessment.capital_summary,
+        maximum_subject_matter_of_dispute_disregard: maximum_subject_matter_of_dispute_disregard(assessment),
+        pensioner_capital_disregard: pensioner_capital_disregard(assessment),
+      )
+    end
```

Comment:
I'm not sure how these 2 methods help? Also I would like to think that at some point the 'assessment' isn't available at this level

Reply:
It's a readability thing. `.call` is getting really hard to understand, so I'm beginning to separate it out into different levels of abstraction. `.call` handles the what, and it passes the how onto the methods it calls.

Agreed, at some point the assessment won't be available at this level - all we need is the capital (and the disposable income data required to calculate the pensioner capital disregard). But that's beyond the scope of this ticket.